### DISCUSSION TITLE
Setting up Mountain Duck fails

### DISCUSSION THREAD

**Discussion by JochenNBV** [iterate-ch/cyberduck#16424] [Q&A] [answered]
Hi,
 
I’d really like to use (and buy) Mountain Duck (“MD”) to replace the awkward OneDrive app on my Mac/our company Macs.
 
Currently I use the test version, free for two weeks. The problem is that I cannot mount anything within OneDrive or SharePoint without being asked for an admin account (either a Microsoft 365 admin account or a Mac admin account, I am not sure which is meant). If I do enter the MS 365 admin credentials, which I have, the MS web interface keeps insisting on an admin account over an over again without connecting to or mounting anything.
 
I’d like to recommend MD to all our users, but it would have to installed and used (inkluding mounting SharePoint directories and more) without having access to a Microsoft 365 admin account. Users should be able to set this up for themselves.
 
We as a company use no Windows machines, but we are using MS 365. All the backend data is handled with Microsoft SharePoint/OneDrive, all the front ends are Macs running macos.
 
Please note: What is also strange: When trying to set up a new connection within MD, most of the fields are greyed out and cannot be filled with anything useful:
<img width="520" alt="image" src="https://github.com/user-attachments/assets/04095166-1e0a-414b-9994-da7d4aa91a29">

Could anyone please give some hints to get MD up and running.
 
Thanks!
 
Regards
Jochen

**Comment by AliveDevil** [comment]
The Microsoft Graph connections (OneDrive, Sharepoint) use the OAuth protocol for user authentication and authorization. The initial setup and approval of the Mountain Duck OAuth application should be performed only once, unless your AD requires, that before a user is added to an OAuth application an admin approval is required (i.e. first login requires approval, subsequent ones don't).
That's also the reason, why most fields are greyed out/disabled:
- Username: Automatically filled through OAuth, and persists for same bookmark
- Password: Doesn't apply for OAuth protocols
- SSH Key: Doesn't apply for anything other than SFTP/SSH
- Client Certificate: Doesn't apply
- Timezone: Doesn't apply
- Timezone: Doesn't apply

You can grant administrator approval for your entire tenant for Mountain Duck, using these steps:
- Open [Microsoft Entra](https://entra.microsoft.com)
- Expand Identity, expand Applications, select Enterprise Application
- Find "Mountain Duck" and open
- In Security-group, select "Permissions"
- Grant Administrator approval

This way all users in your tenant are granted access to use Mountain Duck without requiring initial admin approval.

Alternatively, you can change the [User consent settings](https://entra.microsoft.com/#view/Microsoft_AAD_IAM/ConsentPoliciesMenuBlade/~/UserSettings) in Microsoft Entra to "Allow user consent for apps from verified publishers, for selected permissions (Recommended)", and select following permissions as "low impact":
- Files.ReadWrite.All
- GroupMember.Read.All
- offline_access
- Sites.ReadWrite.All
- User.Read
- or, enable "Allow user consent for apps" accepting the risk this could pose.

If you could share a screenshot of one login attempt to your initial support request (MD-22351), where the admin approval is requested multiple times in a row would be helpful.

**Comment by JochenNBV** [comment]
Hi Jöran,
first of all thanks very much for the swift response. I tried your method to grant admin access from within entra. But after asking for a username with M365 admin rights the web interface for OAuth keeps returning to this screen:
![Bildschirmfoto 2024-10-15 um 12 17 54](https://github.com/user-attachments/assets/8dd4ce32-703b-452c-9a77-4f24ed9cb779)

...without actually **setting** the permission. My Password or 2FA credentials are not asked because in this Chrome profile window these are already stored probably.

So even though your hint was very useful I am unfortunately still stuck in the middle of the process.

**Reply by AliveDevil** [reply]
Thanks, let's continue from there.
A quick search for that prompt leads to [Unexpected error when performing consent to an application](https://learn.microsoft.com/en-us/entra/identity/enterprise-apps/application-sign-in-unexpected-user-consent-error#requesting-not-authorized-permissions-error).

Quoted from there:
> This error can also occur when a user is prevented from consenting to an application due to Microsoft detecting that the permissions request is risky. In this case, an audit event will also be logged with a Category of "ApplicationManagement", Activity Type of "Consent to application" and Status Reason of "Risky application detected".
> 
> Another scenario in which this error might occur is when the user assignment is required for the application, but no administrator consent was provided. In this case, the administrator must first provide tenant-wide admin consent for the application.

Can you check in Entra for the Mountain Duck application Properties, that "Enabled for users to sign-in?" is set to true, and "Assignment required" is disabled.
Do you have anything configured in terms of "Conditional Access"? Had some requests in the past, but never received responses here.

**Comment by JochenNBV** [comment]
Hi Jöran,
thanks again for tryin to help.

This is what Entra says in terms of "Enabled for users to sign-in?" (I have set the UI to German, but this should be what you are looking for):
![Bildschirmfoto 2024-10-16 um 10 56 13](https://github.com/user-attachments/assets/aa35947f-71a4-4028-a744-cfc0c8468824)

The settings seem to be right. but most of it is greyed out which suggest to me that my "second grade only" admin access to our tenant might not be enough. Our IT service provider has the ultimate priviledges...

Also this:

![image](https://github.com/user-attachments/assets/fde552cb-7d4a-456a-ba0c-b942928acc16)

... suggests that I need an action by a higher level of Admin.

And no, nothing is set concerning Conditional Access.

Thanks again for your help which is very much appreciated.
Regards
Jochen

**Reply by AliveDevil** [reply]
Let me check one thing. We are requesting GroupMember.Read.All permissions which doesn't look like it is needed, and is the only OAuth permissions that requires admin consent.

**Reply by AliveDevil** [reply]
If you want, you can navigate to `%ProgramFiles%\Mountain Duck\Profiles` and edit "Microsoft SharePoint.cyberduckprofile".
Remove the line which contains "GroupMember.Read.All". Restart Mountain Duck (if not already closed).
Does this allow you to login?

**Comment by JochenNBV** [comment]
Thanks. Do you a Mac comparable path for me?

**Reply by AliveDevil** [reply]
Sorry, forgot that you used Macs exclusively.

You can navigate to `/Applications/Mountain Duck.app/Contents/Resources/Profiles` and edit `Microsoft SharePoint.cyberduckprofile` with e.g. TextEdit.

**Comment by JochenNBV** [comment]
Great, thanks, this profile editing with TextEdit did the trick. Now I’ll give it a try. I only have one week for testing left, but let’s see.

Thanks again for the really outstanding level of support from your side!!

Regards
Jochen

**Comment by AliveDevil** [comment, marked as answer]
Fix in [^1]. Available in current [snapshot builds of Cyberduck](https://docs.cyberduck.io/cyberduck/preferences/#snapshot-builds) & Mountain Duck.

[^1]: #16435
